pkg/dsdk/util.go

```diff
@@ -368,3 +370,19 @@ func init() {
 	log.SetFormatter(&LogFormatter{})
 	log.SetLevel(log.DebugLevel)
 }
+
+func formatQueryParams(gro *greq.RequestOptions, v reflect.Value, t reflect.Type) {
+	// Formats the Query Params of the Request Option to include
+	// all the fields (name - value) as query params in the URL
+	numFields := t.NumField()
+	params := make(map[string]string)
+	for i := 0; i < numFields; i++ {
+		if t.Field(i).Name == "Ctxt" {
+			continue
+		}
+		key := t.Field(i).Tag.Get("mapstructure")
+		ifc := fmt.Sprintf("%v", v.Field(i).Interface())
```

Comment:
I think v.Field(i).Interface() would fail when reflect.ValueOf() returns Zero value.

Reply:
I believe reflect.ValueOf() can be Zero only if the argument is nil. I could add a nil check for ro in the calling method, but that really means someone passed a nil to the top level SDK API. https://play.golang.org/p/3mVN2UYhCV5